spec/lib/whitehall_importer/create_revision_spec.rb

```diff
@@ -107,14 +117,23 @@
           :whitehall_export_edition,
           attachments: [
             build(:whitehall_export_file_attachment, filename: "attach.txt"),
-            build(:whitehall_export_file_attachment, filename: "attach.txt"),
+            build(:whitehall_export_file_attachment, filename: "subdir/attach.txt"),
           ],
         )
-        revision = described_class.call(document, whitehall_edition)
+        revision = described_class.call(document_import, whitehall_edition)
 
         expect(revision.file_attachment_revisions.first.blob_revision.filename).to eq("attach.txt")
         expect(revision.file_attachment_revisions.last.blob_revision.filename).to eq("attach-1.txt")
       end
+
+      it "skips any attachment it has encountered before" do
+        attachment = build(:whitehall_export_file_attachment, filename: "attach.txt")
+        revision1 = described_class.call(document_import, build(:whitehall_export_edition, attachments: [attachment]))
+        revision2 = described_class.call(document_import, build(:whitehall_export_edition, attachments: [attachment]))
+        expect(revision1.file_attachment_revisions.count).to eq(1)
+        expect(revision2.file_attachment_revisions.count).to eq(1)
+        expect(revision1.file_attachment_revisions.first).to eq(revision2.file_attachment_revisions.first)
```

Comment:
Similar point about whether there's a different assertion we can make that these are effectively equal.